### PR TITLE
Fix auth

### PR DESCRIPTION
**Pull request by wmontwe**
### :tophat: What is the goal?

Auth issue is fixed by using HC Auth SDK 1.16.0 and supporting changes to the SDK

#### Change 

- Android target SDK 31 -> 32
- Change client implementation from Java to Kotlin to aid integration project

#### Removed

* jApiCmp Gradle plugin as it's not needed anymore
* test suite project as it is outdated, alternative will be available in the [integration project](https://github.com/d4l-data4life/hc-sdk-kmp-integration/)

#### Bumped

* Android Gradle Plugin 7.1.3 -> 7.2.0
* Kotlin 1.6.10 -> 1.6.21
* Kotlin Coroutines 1.6.0 -> 1.6.2
* Material 1.5.0 -> 1.6.1
* AndroidX ConstraintLayout 2.1.3 -> 2.1.4
* Spotless Gradle plugin 6.4.2 -> 6.5.1
* Ktlint 0.44.0 -> 0.45.2
* AndroidX AppCompat 1.4.1 -> 1.4.2
* *_BREAKING_* HC Auth SDK 1.15.0 -> 1.16.0

### :thinking: DOD Checklist

- [x] My code follows the code style and naming conventions of this project.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the documentation.
- [x] I have updated the changelog accordingly.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -99,9 +99,9 @@ Once you have a partner ID, follow these steps to start working with the SDK:
 
 === Software requirements
 
-* Android 6.0 (API 23) to Android 11 (API 31)
+* Android 6.0 (API 23) to Android 12L (API 32)
 * Java 8 link:https://developer.android.com/studio/write/java8-support[Limitations] link:https://jakewharton.com/d8-library-desugaring/[Desugaring]
-* Kotlin 1.6.10
+* Kotlin 1.6.21
```
Comment (bitPogo) [inline README.adoc:104]: Maybe use 1.7.0? Compose 1.2.0 should support it.

**Review by bitPogo — CHANGES_REQUESTED** [review]
Some minor suggestions

File: README.adoc (hunk 1 above)
Comment (wmontwe) [reply README.adoc:104]: Should be done in another pull-request

File: buildSrc/build.gradle.kts [hunk 2]
```diff
@@ -53,8 +53,8 @@ dependencies {
     // publishing.gradle.kts
     implementation("org.eclipse.jgit:org.eclipse.jgit:6.1.0.202203080745-r")
     // quality.gradle.kts
-    implementation("com.diffplug.spotless:spotless-plugin-gradle:6.4.2")
-    implementation("com.pinterest:ktlint:0.44.0")
+    implementation("com.diffplug.spotless:spotless-plugin-gradle:6.5.1")
+    implementation("com.pinterest:ktlint:0.45.2")
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply buildSrc/build.gradle.kts:57]: see above

File: buildSrc/src/main/kotlin/care/data4life/gradle/core/config/D4LClientConfig.kt [hunk 3]
```diff
@@ -16,13 +16,31 @@
...
+            "clientSecret" to get(environment).secret,
+            "redirectScheme" to get(environment).redirectScheme,
+        ).also {
+            if (environment == PRODUCTION && debug == null) {
+                it["debug"] = "false"
+            } else if (debug != null) {
+                it["debug"] = debug.toString()
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply buildSrc/src/main/kotlin/care/data4life/gradle/core/config/D4LClientConfig.kt:43]: thanks :-)

File: sdk-jvm/src/main/kotlin/care/data4life/sdk/UserIdExtractor.kt [hunk 4]
```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2020 D4L data4life gGmbH / All rights reserved.
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply sdk-jvm/src/main/kotlin/care/data4life/sdk/UserIdExtractor.kt:2]: Right

File: sdk-jvm/src/main/kotlin/care/data4life/sdk/UserIdExtractor.kt [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        val decodedUrl = urlDecoder.decode(stateString)
+        val dataString = base64.decodeToString(decodedUrl)
+        val state: AuthState?
+        try {
+            state = adapter.fromJson(dataString)
+            return state!!.alias
+        } catch (e: IOException) {
+            // ignore
+        }
+        return null
```
[the remark this answers is not in the document]
Comment (wmontwe) [reply sdk-jvm/src/main/kotlin/care/data4life/sdk/UserIdExtractor.kt:53]: good catch!

**Review by bitPogo — APPROVED** [review]
LGTM

**Pull request merged**